File: Sierra-Code-Platoon/Week1/Day 1/algo-linear-search/python/linear_search.py
from array import array


array_to_search_through = []
for number in range(1, 1001):
    array_to_search_through.append(number)

def linear_search(value_to_find, array_to_search_through):
    # your code here
    
    for x in range(len(array_to_search_through)):
        if(array_to_search_through[x] == value_to_find):
            return x
    
    return None

def linear_search_global(value_to_find, array_to_search_through):
    # your code here
    result = []
    
    for x in range(len(array_to_search_through)):
        if(array_to_search_through[x] == value_to_find):
            result.append(x)
    
    if(len(result) == 0):
        return None
    
    return result

File: Sierra-Code-Platoon/Week1/Day 1/algo-linear-search/js/linearSearch.js
const arrayToSearchThrough = [];
for (let i = 1; i <= 5; i++) {
    arrayToSearchThrough.push(i);
}
//  console.log(arrayToSearchThrough);
exports.linearSearch = function(valueToFind, arrayToSearchThrough) {
    
    for(let index = 0; index < arrayToSearchThrough.length; index++){
        console.log(index);
        if(valueToFind == arrayToSearchThrough[index]){
            return index;
        }
    };
    return undefined
}

// console.log(linearSearch(2, arrayToSearchThrough));

// exports.linearSearchGlobal = function(valueToFind, arrayToSearchThrough) {

exports.linearSearchGlobal = function (valueToFind, arrayToSearchThrough) {
    // your code here

    var result = [];

    for(let index = 0; index < arrayToSearchThrough.length; index++){
        if(arrayToSearchThrough[index] == valueToFind){
            result.push(index);
        }
    }
    
    if(result.length == 0){
        return undefined;
    }else{
        return result;
    }
    
};

// console.log(linearSearchGlobal('a',['b','a','n','a','n','a']));